arch/arm/core/aarch32/thread.c

```diff
@@ -128,13 +128,25 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 #if defined(CONFIG_USERSPACE)
 	thread->arch.priv_stack_start = 0;
 #endif
+#endif
+
+#if defined(CONFIG_USE_SWITCH)
+	thread->switch_handle = (void *)thread;
 #endif
 	/*
 	 * initial values in all other registers/thread entries are
 	 * irrelevant.
 	 */
 }
 
+#if defined(CONFIG_USE_SWITCH)
+void *z_arch_get_next_switch_handle(struct k_thread **old_thread)
```

Comment:
Nitpick: I know this naming is preexisting (I think arc did it first, with sparc and arm64 getting it via osmosis?), but it's a little confusing.  This isn't a function provided by the arch layer to implement "get_next_switch_handle", it's a **wrapper** being used internal to the architecture to save off the current thread pointer prior to the context switch in an interrupt handler.  "z_get_next_switch_handle()" is in fact a portable (not arch-layer) scheduler function that just returns the context to which the ISR should return.

I'd suggest renaming this rather than continuing the cut/paste: "arm32_next_switch_handle()" or something?

Reply:
I don't know where the chain starts when it comes to this particular function's naming, but I did in fact extend it from arm64 to arm. I'll change the name of this wrapper function in order to resolve that confusion.